packages/xo-server/src/api/vm.js

```diff
@@ -480,24 +482,29 @@ export async function migrate({
 
   await this.checkPermissions(this.user.id, permissions)
 
-  await this.getXapi(vm).migrateVm(
-    vm._xapiId,
-    this.getXapi(host),
-    host._xapiId,
-    {
+  await this.getXapi(vm)
+    .migrateVm(vm._xapiId, this.getXapi(host), host._xapiId, {
       sr: sr && this.getObject(sr, 'SR')._xapiId,
       migrationNetworkId:
         migrationNetwork != null ? migrationNetwork._xapiId : undefined,
       mapVifsNetworks: mapVifsNetworksXapi,
       mapVdisSrs: mapVdisSrsXapi,
-    }
-  )
+      force,
+    })
+    .catch(error => {
+      if (error?.code !== undefined) {
+        throw operationFailed({ objectId: vm.id, code: error.code })
+      }
+      throw error
+    })
```

Comment:
This change cause problems because it hides the original errors.

See xoa-support#2907